### PR TITLE
add service accounts for all functions

### PR DESCRIPTION
**Pull request by vranystepan**
just adding remaining Service Accounts. Not yet applied. I'm not really sure about `register-notification-aftermath` -> `reg-notification-aftermath` (28 chars limit) 

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.erouska.google_service_account.changepushtoken will be created
  + resource "google_service_account" "changepushtoken" {
      + account_id   = "change-push-token"
      + display_name = "ChangePushToken cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.checkattemptsthresholds will be created
  + resource "google_service_account" "checkattemptsthresholds" {
      + account_id   = "check-attempts-thresholds"
      + display_name = "CheckAttemptsThresholds cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.increaseehridattemptscount will be created
  + resource "google_service_account" "increaseehridattemptscount" {
      + account_id   = "increase-ehrid-attempts-count"
      + display_name = "IncreaseEhridAttemptsCount cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.increasenotificationscounter will be created
  + resource "google_service_account" "increasenotificationscounter" {
      + account_id   = "increase-notifications-counter"
      + display_name = "IncreaseNotificationsCounter cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.isehridactive will be created
  + resource "google_service_account" "isehridactive" {
      + account_id   = "is-ehrid-active"
      + display_name = "IsEhridActive cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.provideverificationcode will be created
  + resource "google_service_account" "provideverificationcode" {
      + account_id   = "provide-verification-code"
      + display_name = "ProvideVerificationCode cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registerehrid will be created
  + resource "google_service_account" "registerehrid" {
      + account_id   = "register-ehrid"
      + display_name = "RegisterEhrid cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registernotification will be created
  + resource "google_service_account" "registernotification" {
      + account_id   = "register-notification"
      + display_name = "RegisterNotification cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registernotificationaftermath will be created
  + resource "google_service_account" "registernotificationaftermath" {
      + account_id   = "reg-notification-aftermath"
      + display_name = "RegisterNotificationAfterMath cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 9 to add, 0 to change, 0 to destroy.
```

### CODE REVIEW

**Comment by vranystepan** [comment]
```
Terraform will perform the following actions:

  # module.erouska.google_project_iam_member.calculatecoviddata[0] will be destroyed
  - resource "google_project_iam_member" "calculatecoviddata" {
      - etag    = "BwWvS26JUio=" -> null
      - id      = "erouska-key-server-dev/roles/cloudfunctions.serviceAgent/serviceaccount:calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - member  = "serviceAccount:calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - project = "erouska-key-server-dev" -> null
      - role    = "roles/cloudfunctions.serviceAgent" -> null
    }

  # module.erouska.google_project_iam_member.calculatecoviddata[1] will be destroyed
  - resource "google_project_iam_member" "calculatecoviddata" {
      - etag    = "BwWvS26JUio=" -> null
      - id      = "erouska-key-server-dev/roles/datastore.user/serviceaccount:calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - member  = "serviceAccount:calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - project = "erouska-key-server-dev" -> null
      - role    = "roles/datastore.user" -> null
    }

  # module.erouska.google_project_iam_member.changepushtoken[0] will be created
  + resource "google_project_iam_member" "changepushtoken" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.changepushtoken[1] will be created
  + resource "google_project_iam_member" "changepushtoken" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.user"
    }

  # module.erouska.google_project_iam_member.isehridactive[0] will be created
  + resource "google_project_iam_member" "isehridactive" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.isehridactive[1] will be created
  + resource "google_project_iam_member" "isehridactive" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.viewer"
    }

  # module.erouska.google_project_iam_member.provideverificationcode[0] will be created
  + resource "google_project_iam_member" "provideverificationcode" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.provideverificationcode[1] will be created
  + resource "google_project_iam_member" "provideverificationcode" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.viewer"
    }

  # module.erouska.google_project_iam_member.registerehrid[0] will be created
  + resource "google_project_iam_member" "registerehrid" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.registerehrid[1] will be created
  + resource "google_project_iam_member" "registerehrid" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/iam.serviceAccountTokenCreator"
    }

  # module.erouska.google_project_iam_member.registerehrid[2] will be created
  + resource "google_project_iam_member" "registerehrid" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.user"
    }

  # module.erouska.google_project_iam_member.registernotification[0] will be created
  + resource "google_project_iam_member" "registernotification" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.registernotification[1] will be created
  + resource "google_project_iam_member" "registernotification" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.user"
    }

  # module.erouska.google_project_iam_member.registernotificationaftermath[0] will be created
  + resource "google_project_iam_member" "registernotificationaftermath" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/cloudfunctions.serviceAgent"
    }

  # module.erouska.google_project_iam_member.registernotificationaftermath[1] will be created
  + resource "google_project_iam_member" "registernotificationaftermath" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = (known after apply)
      + role    = "roles/datastore.user"
    }

  # module.erouska.google_service_account.calculatecoviddata will be destroyed
  - resource "google_service_account" "calculatecoviddata" {
      - account_id   = "calculate-covid-data-increase" -> null
      - display_name = "CalculateCovidDataIncrease cloud function service account" -> null
      - email        = "calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - id           = "projects/erouska-key-server-dev/serviceAccounts/calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - name         = "projects/erouska-key-server-dev/serviceAccounts/calculate-covid-data-increase@erouska-key-server-dev.iam.gserviceaccount.com" -> null
      - project      = "erouska-key-server-dev" -> null
      - unique_id    = "106804147902781807780" -> null
    }

  # module.erouska.google_service_account.changepushtoken will be created
  + resource "google_service_account" "changepushtoken" {
      + account_id   = "change-push-token"
      + display_name = "ChangePushToken cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.isehridactive will be created
  + resource "google_service_account" "isehridactive" {
      + account_id   = "is-ehrid-active"
      + display_name = "IsEhridActive cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.provideverificationcode will be created
  + resource "google_service_account" "provideverificationcode" {
      + account_id   = "provide-verification-code"
      + display_name = "ProvideVerificationCode cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registerehrid will be created
  + resource "google_service_account" "registerehrid" {
      + account_id   = "register-ehrid"
      + display_name = "RegisterEhrid cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registernotification will be created
  + resource "google_service_account" "registernotification" {
      + account_id   = "register-notification"
      + display_name = "RegisterNotification cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # module.erouska.google_service_account.registernotificationaftermath will be created
  + resource "google_service_account" "registernotificationaftermath" {
      + account_id   = "reg-notification-aftermath"
      + display_name = "RegisterNotificationAfterMath cloud function service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 19 to add, 0 to change, 3 to destroy.
```

File: terraform/modules/erouska/coviddata.tf [hunk 1]
```diff
@@ -10,15 +10,41 @@ locals {
...
+  getcoviddata_roles = [
+    "roles/cloudfunctions.serviceAgent",
+    "roles/datastore.viewer"
+  ]
+
+  changepushtoken_roles = [
     "roles/cloudfunctions.serviceAgent",
     "roles/datastore.user"
   ]
-  getcoviddata_roles = [
```
Comment (marekaf) [inline terraform/modules/erouska/coviddata.tf:17]: why is this deleted? GetCovidData is still there...

Comment (vranystepan) [reply terraform/modules/erouska/coviddata.tf:17]: is it deleted? https://github.com/covid19cz/erouska-backend/blob/sv-iam-service-accounts/terraform/modules/erouska/coviddata.tf#L13

Comment (marekaf) [reply terraform/modules/erouska/coviddata.tf:17]: ahaa! sorry, just the github diff didn't manage to pick up the change!
